src/containers/StructurePage/StructureContainer.jsx

```diff
@@ -360,13 +368,44 @@ export class StructureContainer extends React.PureComponent {
     });
   }
 
+  toggleFavorite(subjectId) {
+    let cookie = getCookie('favorite_subjects', document.cookie);
+    let favorites = [].concat(JSON.parse(cookie));
+
+    if (favorites.includes(subjectId)) {
+      favorites = favorites.filter(id => id !== subjectId);
+    } else {
+      favorites.push(subjectId);
+    }
+
+    setCookie('favorite_subjects', JSON.stringify(favorites), false);
+
+    this.forceUpdate();
+  }
+
+  fetchFavoritesSubjectIds() {
+    return [].concat(
+      JSON.parse(getCookie('favorite_subjects', document.cookie)),
+    );
+  }
+
+  fetchFavoriteSubjects(subjects, favoriteSubjectIds) {
+    return subjects.filter(e => favoriteSubjectIds.includes(e.id));
+  }
+
+  toggleShowFavorites() {
+    const { showFavorites } = this.state;
+    this.setState({ showFavorites: !showFavorites });
+  }
```

Comment:
Samme som over med at vi burde bruke `prevState` når vi bruker tidligere state for å endre staten.
```suggestion
  toggleShowFavorites() {
    this.setState(prevState => { showFavorites: !prevState.showFavorites });
  }
```